fedimint-api/src/config.rs

```diff
@@ -614,26 +613,30 @@ impl<G: DkgGroup> Dkg<G> {
     }
 
     /// Runs a single step of the DKG algorithm, processing a `msg` from `peer`
-    pub fn step(&mut self, peer: PeerId, msg: DkgMessage<G>) -> DkgStep<G> {
+    pub fn step(&mut self, peer: PeerId, msg: DkgMessage<G>) -> anyhow::Result<DkgStep<G>> {
         match msg {
             DkgMessage::HashedCommit(hashed) => {
                 match self.hashed_commits.get(&peer) {
-                    Some(old) if *old != hashed => panic!("{peer} sent us two hashes!"),
+                    Some(old) if *old != hashed => {
+                        return Err(format_err!("{peer} sent us two hashes!"))
+                    }
                     _ => self.hashed_commits.insert(peer, hashed),
                 };
 
                 if self.hashed_commits.len() == self.peers.len() {
                     let our_commit = self.commitments[&self.our_id].clone();
-                    return self.broadcast(DkgMessage::Commit(our_commit));
+                    return Ok(self.broadcast(DkgMessage::Commit(our_commit)));
                 }
             }
             DkgMessage::Commit(commit) => {
                 let hash = self.hash(commit.clone());
-                assert_eq!(self.threshold, commit.len(), "wrong degree from {peer}");
-                assert_eq!(hash, self.hashed_commits[&peer], "wrong hash from {peer}");
+                ensure!(self.threshold == commit.len(), "wrong degree from {peer}");
+                ensure!(hash == self.hashed_commits[&peer], "wrong hash from {peer}");
```

Comment:
Is this array access guaranteed to always be in bounds? `.get` is generally preferable imo since it's more apparent if something can fail.